### PR TITLE
New devtools registration mechanism

### PR DESCRIPTION
**Pull request by phryneas**
This is the counter PR to https://github.com/apollographql/apollo-client-devtools/pull/1038

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

File: src/core/ApolloClient.ts [hunk 1]
```diff
@@ -199,30 +205,32 @@ export class ApolloClient<TCacheShape> implements DataProxy {
...
-        const nav = window.navigator;
-        const ua = nav && nav.userAgent;
-        let url: string | undefined;
-        if (typeof ua === "string") {
-          if (ua.indexOf("Chrome/") > -1) {
-            url = "https://chrome.google.com/webstore/detail/" +
-              "apollo-client-developer-t/jdkknkkbebbapilgoeccciglkfbmbnfm";
-          } else if (ua.indexOf("Firefox/") > -1) {
-            url = "https://addons.mozilla.org/en-US/firefox/addon/apollo-developer-tools/";
+      setTimeout(() => {
```
Comment (phryneas) [inline src/core/ApolloClient.ts:208]: This is the reason for the sometimes-wrong warning: while the devtools wait for seconds for the client to register, the client immediately logs if the devtools are not there yet. => this test needs to be delayed.

**Comment by alessbell** [comment]
Tested this out with the latest (unreleased) devtools build in Chrome/FF

This screenshot _doesn't_ show the beautifully empty console (no erroneous warning about needing to install the devtools!) but it _does_ show off the new syntax highlighting @jerelmiller added that has been thus unreleased (it's using the same colors from our docs, nice touch 😎)

CleanShot 2023-08-30 at 13 37 20

**Comment by alessbell** [comment]
Targeted this at the `release-3.9` branch, though since it's fully backward compatible I don't see a reason not to ship it in a patch. Thoughts @jerelmiller @phryneas?

**Comment by phryneas** [comment]
Yeah, I agree, this should land in a 3.8 patch release :)

Cleaning it up:

> 🚢 it

@jerelmiller is that a green review checkmark? :D

**Review by alessbell — APPROVED** [review]
Excited to get this out in the next patch!

**Pull request merged**